### PR TITLE
prevent submitting duplicate proposals + deepEqualJSONType() to giLodash

### PR DESCRIPTION
**Pull request by taoeffect**
Simple PR that:

- adds `'state/vuex/getters'` selector
- adds fast deepEqual for JSON types `_.deepEqualJSONType()`
- adds check to throw exception if duplicate proposal submitted

### CODE REVIEW

File: frontend/model/contracts/group.js
Comment (sandrina-p) [inline frontend/model/contracts/group.js:143]: even if created by a different member, I don't think it should be allowed...

**Review by sandrina-p — APPROVED** [review]

File: frontend/model/contracts/group.js
Comment (taoeffect) [reply frontend/model/contracts/group.js:143]: I see what you mean, but this is mainly to prevent the kind of bug that I encountered yesterday. It's technically not a bannable offense for someone else to propose the same thing you're proposing. It could be just a mistake

**Comment by taoeffect** [comment]
**NOTE: this is an exceptional merge that violates Rule 7 of our contributions guidelines because it's during a hackathon and I have approval from @sandrina-p to merge it, and @sandrina-p cannot merge it herself because the build is failing, but it's an error unrelated to this PR.**

**Pull request merged**
